sway/commands/gaps.c

```diff
@@ -68,15 +43,19 @@ static struct cmd_results *gaps_set_defaults(int argc, char **argv) {
 		return cmd_results_new(CMD_INVALID, "gaps",
 				"Expected 'gaps inner|outer <px>'");
 	}
-	if (amount < 0) {
-		amount = 0;
-	}
-
 	if (inner) {
-		config->gaps_inner = amount;
+		config->gaps_inner = (amount >= 0) ? amount : 0;
 	} else {
 		config->gaps_outer = amount;
 	}
+
+    // Prevent negative outer gaps from moving windows out of the container.
+    int overflow;
+    overflow = (config->gaps_inner + config->gaps_outer);
+    if (overflow < 0) {
```

Comment:
You could do this without introducing a new variable:

    if (config->gaps_outer < -config->gaps_inner) {
        config->gaps_outer = -config->gaps_inner;
    }

Reply:
Done